dashboard/modules/job/job_manager.py

```diff
@@ -411,15 +418,21 @@ async def run(
             ray.actor.exit_actor()
 
     def stop(self):
-        """Set step_event and let run() handle the rest in its asyncio.wait()."""
+        """Set stop_event and let run() handle the rest in its asyncio.wait()."""
         self._stop_event.set()
 
+    def delete(self):
```

Comment:
what is the difference for `stop` and `delete`? (looks like they both make the job stop immediately)

Reply:
Here the main point is that `delete` deletes the job info and status from internal storage.  The initial idea was to have `delete` also stop the job, but as Ed suggested maybe the best thing to do is to only allow deleting jobs that have already been stopped.

Reply:
@sihanwang41 I confirmed that `internal_kv_delete` can be called concurrently, so we don't need to use a lock at the application level.